src/main/java/eod/warObject/WarObject.java

```diff
@@ -0,0 +1,74 @@
+package eod.warObject;
+
+
+import eod.GameObject;
+import eod.Gameboard;
+import eod.Player;
+
+import java.awt.*;
+import java.util.ArrayList;
+
+//WarObject represented anything on the gameboard
+public abstract class WarObject implements GameObject {
+    public Point position;
+    protected Player player;
+
+    public WarObject(Player player, Point point) {
+        this.player = player;
+        position = point;
+    }
```

Comment:
WarObject should be created before putting on the board, however, it's impossible to know the position of the object before it is put on the board.
I would suggest removing position as a parameter.